src/sentry/static/sentry/app/components/forms/selectControl.jsx

```diff
@@ -286,6 +286,7 @@ const SelectControl = props => {
       value={mappedValue}
       isMulti={props.multiple || props.multi}
       isDisabled={props.isDisabled || props.disabled}
+      defaultOptions={choicesOrOptions}
```

Comment:
@mgaeta I don't think the fix is to change the select control. The `SentryAppExternalIssueForm` component is able to have this functionality work without this change. We should change the actual form instead I think.